doc/Makefile

```diff
@@ -17,18 +12,14 @@ $(error The '$(SPHINXBUILD)' command was not found. Make sure you have Sphinx in
 endif
 
 # Internal variables.
-PAPEROPT_a4     = -D latex_paper_size=a4
-PAPEROPT_letter = -D latex_paper_size=letter
-ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
+allsphinxopts   = -d $(BUILDDIR)/doctrees $(SPHINXOPTS) .
```

Comment:
why rename just this var to lowercase?

Reply:
@andrewboie 
See https://github.com/zephyrproject-rtos/zephyr/pull/8288/commits/7698de922a16b7aa35d727930528bf2b08d1d45c.

Seen that idiom in a bunch of places, and I think it makes Makefiles a lot clearer. It is mentioned in the GNU Make [manual](https://www.gnu.org/software/make/manual/html_node/Using-Variables.html) as well:

```
It is traditional to use upper case letters in variable names, but we recommend using lower case letters for variable names that serve internal purposes in the makefile, and reserving upper case for parameters that control implicit rules or for parameters that the user should override with command options (see Overriding Variables).
```